src/firefly/js/ui/FieldGroupEnable.jsx

```diff
@@ -0,0 +1,116 @@
+}
+
+
+
+
+/**
+ * @global
+ * @public
+ * @typedef {Object} ConnectorInterface
+ *
+ * @summary Return value of useFieldGroupConnector. This object contains the view components properties (viewProps) and the
+ * fireValueChange. Simple pass the viewProps to the component and call fireValueChange when the view component has a
+ * value change call.
+ *
+ * @prop {Function} fireValueChange- Call the anytime the value of the view changes. The parameter is an object
+ * the should contains as least a value properties plus anything else that is appropriate to pass
+ * @prop {Object} viewProps - all the properties passed to useFieldGroupConnector with the connection properties
+ * removed.  This object should contain all the properties to pass on the the view component
+ */
+
+
+
+/**
+ *
+ * Hook to connect a field to the FieldGroup Store. Pass the props object, make sure it includes the required props
+ * to connect to the store (fieldKey is the only requirement, see below). The hooks returns qn object with the new
+ * props that you should be able to pass directly to the view.
+ *
+ * the props object parameter can contain any that should be kept in the store. The parameters below are special.
+ * fieldKey is required.
+ *
+ * @param {Object} props
+ * @param {string} props.fieldKey - required, a unique id for this field (unique within group)
+ * @param {string} [props.groupKey] - optional - a unique group id, normally this is not use because it is passed in the context
+ * @param {string} [props.forceReinit] - optional - if true, this field will be reinited from the properties and not from the field group
+ * @param {string} [props.initialState] - optional - the initial state object
+ * @param {function} [props.confirmInitialValue] - optional - on the first render the value from the properties is passed and a value is returned
+ * @return {ConnectorInterface}
+ *
+ */
+export const useFieldGroupConnector= (props) => {
+    const infoRef = useRef({prevFieldKey:undefined, prevGroupKey:undefined});
+    const gkFromCtx= useContext(GroupKeyCtx);
+    const {fieldKey, forceReinit, confirmInitialValue= defaultConfirmInitValue, initialState}= props;
+    const groupKey= props.groupKey || gkFromCtx;
+
+    function getInitialState() {
+        const storeField= get(FieldGroupUtils.getGroupFields(groupKey), [fieldKey]);
+        const initS= forceReinit ? (initialState ||  storeField || {}) : (storeField || initialState || {});
+        return {...initS, value: confirmInitialValue(initS.value,props,initS)};
+    }
+    const [fieldState, setFieldState] = useState(getInitialState());
+
+
+    useEffect(() => {
+        const {prevFieldKey, prevGroupKey}= infoRef.current;
+        if (prevFieldKey!==fieldKey || prevGroupKey !== groupKey) {  // called the first time or when fieldKey or groupKey change
+            let value= fieldState.value;
+            if (prevFieldKey && prevGroupKey) { // if field and group key changed, whole thing must reinit
```

Comment:
I may not understand this right.  But, if the Component is keyed by `groupKey + fieldKey`, then this will happen without any logic.  It looks like you're trying to reuse the component when a new one could be created instead.

Reply:
the problem is  that the properties can change on a component with out is going away.  This is a react thing.  We deal with in in the HOC now, but it is a log less readable.

Reply:
The problem is that the properties can change on a component without it going away. This is a react thing. We deal with this now with the HOC now, but it is a log less readable.